src/serializer/ResidualHeapSerializer.js

```diff
@@ -1674,13 +1674,25 @@ export class ResidualHeapSerializer {
     let remainingProperties = new Map(val.properties);
     const dummyProperties = new Set();
     let props = [];
+    let isFunction = f => f instanceof ECMAScriptSourceFunctionValue;
+    let isLeaked = val.isLeakedObject();
+
+    let shouldDropAsAssignedProp = (descriptor: Descriptor) =>
+      isLeaked && !(isFunction(descriptor.get) || isFunction(descriptor.set));
```

Comment:
This special case for accessor properties here (and the corresponding code in the visitor) only exists because of this comment `// TODO: Deal with accessor properties` in the materialization code, right? 

If so, please create an issue, and have comments like `TODO #Issue` in all the relevant places in the code.

Reply:
Done. Yes, things get complicated with getters and setters, especially if they cross reference other properties in the object.